src/comestible_inv.cpp

```diff
@@ -0,0 +1,2477 @@
+                                           charges_total ) );
+            } else {
+                item_name = it.display_money( sitem.items.size(), charges_total );
+            }
+        } else {
+            if( stolen ) {
+                item_name = string_format( "%s %s", stolen_string, it.display_name() );
+            } else {
+                //item_name = it.display_name();
+                item_name = sitem.items.front()->display_name();
+            }
+        }
+        if( get_option<bool>( "ITEM_SYMBOLS" ) ) {
+            item_name = string_format( "%s %s", it.symbol(), item_name );
+        }
+
+        //print item name
+        trim_and_print( window, 6 + x, compact ? 1 : 4, max_name_length, thiscolor, item_name );
+
+        //print src column
+        // TODO: specify this is coming from a vehicle!
+        if( pane.get_area() == AIM_ALL && !compact ) {
+            mvwprintz( window, 6 + x, src_startpos, thiscolor, squares[sitem.area].shortname );
+        }
+
+        //print "amount" column
+        int it_amt = sitem.stacks;
+        if( it_amt > 1 ) {
+            print_color = thiscolor;
+            if( it_amt > 9999 ) {
+                it_amt = 9999;
+                print_color = selected ? hilite( c_red ) : c_red;
+            }
+            mvwprintz( window, 6 + x, amt_startpos, print_color, "%4d", it_amt );
+        }
+
+        //print weight column
+        double it_weight = convert_weight( sitem.weight );
+        size_t w_precision;
+        print_color = it_weight > 0 ? thiscolor : thiscolordark;
+
+        if( it_weight >= 1000.0 ) {
+            if( it_weight >= 10000.0 ) {
+                print_color = selected ? hilite( c_red ) : c_red;
+                it_weight = 9999.0;
+            }
+            w_precision = 0;
+        } else if( it_weight >= 100.0 ) {
+            w_precision = 1;
+        } else {
+            w_precision = 2;
+        }
+        mvwprintz( window, 6 + x, weight_startpos, print_color, "%5.*f", w_precision, it_weight );
+
+        //print volume column
+        bool it_vol_truncated = false;
+        double it_vol_value = 0.0;
+        std::string it_vol = format_volume( sitem.volume, 5, &it_vol_truncated, &it_vol_value );
+        if( it_vol_truncated && it_vol_value > 0.0 ) {
+            print_color = selected ? hilite( c_red ) : c_red;
+        } else {
+            print_color = sitem.volume.value() > 0 ? thiscolor : thiscolordark;
+        }
+        mvwprintz( window, 6 + x, vol_startpos, print_color, it_vol );
+
+        if( sitem.autopickup ) {
+            mvwprintz( window, 6 + x, 1, magenta_background( it.color_in_inventory() ),
+                       compact ? it.tname().substr( 0, 1 ) : ">" );
+        }
+        //get_consumable_item
+        char const *string_format = "";
+        islot_comestible it_c = get_edible_comestible( p, it ); // .type->comestible;
+
+        //print "CALORIES" column
```

Comment:
I think it would be nice to have dedicated function to draw column to reduce code duplication.